src/backend/api/Fusion.Resources.Domain/Commands/Requests/InternalRequests/UpdateInternalRequest.cs

```diff
@@ -50,7 +50,7 @@ public Handler(ResourcesDbContext db, IProfileService profileService, IMediator
 
             public async Task<QueryResourceAllocationRequest> Handle(UpdateInternalRequest request, CancellationToken cancellationToken)
             {
-                var dbRequest = await db.ResourceAllocationRequests.FirstAsync(r => r.Id == request.RequestId);
+                var dbRequest = await db.ResourceAllocationRequests.FirstAsync(r => r.Id == request.RequestId, cancellationToken);
```

Comment:
During modification of data/state. Do we really want to be able to halt execution in the middle of whatever happens ?